ml-package-versions.yml

```diff
@@ -0,0 +1,179 @@
+    requirements:
+      # https://github.com/pytorch/vision#installation
+      "== 1.4.0": ["torchvision==0.5.0", "scikit-learn"]
+      "== 1.5.0": ["torchvision==0.6.0", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+  autologging:
+    minimum: "1.4.0"
+    maximum: "1.7.0"
+    requirements:
+      "== 1.4.0": ["torchvision==0.5.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.0": ["torchvision==0.6.0", "pytorch-lightning", "scikit-learn"]
+      "== 1.5.1": ["torchvision==0.6.1", "pytorch-lightning", "scikit-learn"]
+      "== 1.6.0": ["torchvision==0.7.0", "pytorch-lightning", "scikit-learn"]
+      "> 1.6.0": ["torchvision", "pytorch-lightning", "scikit-learn"]
+    run: |
+      pytest tests/pytorch/test_pytorch_autolog.py --large
+
+tensorflow-1.x:
+  package_info:
+    pip_release: "tensorflow"
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow/test_tensorflow_model_export.py --large
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["h5py<3.0"]
+    run: |
+      pytest tests/tensorflow_autolog/test_tensorflow_autolog.py --large
+
+tensorflow-2.x:
+  package_info:
+    pip_release: "tensorflow"
+    install_dev: |
+      pip install tf-nightly
+
+  models:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    run: |
+      pytest tests/tensorflow/test_tensorflow2_model_export.py --large
+
+  autologging:
+    minimum: "2.0.0"
+    maximum: "2.3.1"
+    requirements:
+      "< 2.2": ["h5py<3.0"]
+    run: |
+      pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+
+keras-tf1.x:
+  package_info:
+    pip_release: "tensorflow"
+
+  models:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["keras==2.2.5", "h5py<3.0", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+    run: |
+      pytest tests/keras/test_keras_model_export.py --large
+
+  autologging:
+    minimum: "1.15.4"
+    maximum: "1.15.4"
+    requirements: ["keras==2.2.5", "h5py<3.0"]
+    run: |
+      pytest --verbose tests/keras_autolog/test_keras_autolog.py --large
+
+keras-tf2.x:
+  package_info:
+    pip_release: "keras"
+    install_dev: |
+      # In keras >= 2.4.0, `keras` simply redirects to `tensorflow.keras`:
+      # https://github.com/keras-team/keras#multi-backend-keras-and-tfkeras
+      pip install tf-nightly 'keras>=2.4.0'
+
+  models:
+    minimum: "2.3.1"
+    maximum: "2.4.3"
+    requirements:
+      "== 2.3.1":
+        ["tensorflow==2.2.1", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+      "> 2.3.1": ["tensorflow", "scikit-learn", "pyspark==2.4.0", "pyarrow"]
+    run: |
+      pytest tests/keras/test_keras_model_export.py --large
+
+  autologging:
+    minimum: "2.3.1"
```

Comment:
I think we support Keras 2.0.0 and above. Are we sure this is the correct minimum version?

Reply:
Thanks for the catch. Fixed it and found new errors!

Reply:
In keras 2.16.0, `is_plain_keras` doesn't seem to work.

https://github.com/mlflow/mlflow/blob/7d185a8f777864bd5d7e66735108e457b1a80852/mlflow/keras.py#L168-L175

https://github.com/mlflow/mlflow/pull/3731/checks?check_run_id=1487102127#step:7:2043

Reply:
In keras 2.0.9,

```
ValueError: The first layer in a Sequential model must get an `input_shape` or `batch_input_shape` argument.
```

https://github.com/mlflow/mlflow/pull/3731/checks?check_run_id=1487102127#step:7:114

```
UnboundLocalError: local variable 'arrays' referenced before assignment
```

https://github.com/mlflow/mlflow/pull/3731/checks?check_run_id=1487102103#step:7:1671